src/utils/metrics.ts

```diff
@@ -170,7 +170,8 @@ export async function _insert(
   data: Record<string, any>,
   targetConfig: TargetConfig
 ) {
-  if (process.env.DRY_RUN) {
+  // DRY_RUN ready through env variable will be a string, so we need to be more explicit here.
```

Comment:
Typo:

```suggestion
  // DRY_RUN read through env variable will be a string, so we need to be more explicit here.
```

But also ... really? Empty string is falsey. We're getting something other than that here? 🤔

```
$ node                                                                                                         
Welcome to Node.js v19.9.0.
Type ".help" for more information.
> Boolean('')
false
> Boolean(' ')
true
> 
```

Reply:
Well, I was using `DRY_RUN=0` which wasn't working for me

Reply:
Ah, well, yeah, that makes sense.  :)

If I comment out `DRY_RUN` (don't set it at all) then the test suite passes:

```
Test Suites: 43 passed, 43 total
Tests:       1 skipped, 297 passed, 298 total
Snapshots:   29 passed, 29 total
Time:        18.966 s, estimated 19 s
```

Reply:
I think it doesn't hurt to be more explicit here. Also, adding `DRY_RUN=` to the `.env.test` seems to make the tests pass even if I set `DRY_RUN=1` in my `.env` file

Reply:
> Also, adding `DRY_RUN=` to the `.env.test` seems to make the tests pass even if I set `DRY_RUN=1` in my `.env` file

Makes sense, I guess it's empty string in that case. Nice find.

Reply:
> I think it doesn't hurt to be more explicit here.

Okay but in that case we should move the parsing to `src/config/index.ts` and use that from `metrics.ts`, and we should also support "yes" and we should be case insensitive, and are we doing similar parsing for any other boolean-ish envvars?

Reply:
Yeah, that does sound reasonable. Scanning through I see one example of another boolean env var:
```
export const DISABLE_GITHUB_METRICS =
  process.env.DISABLE_GITHUB_METRICS == 'true';
```

Reply:
> I see one example of another boolean env var

How about we make `DRY_RUN` behave the same as that for consistency with the status quo?

Reply:
> How about we make DRY_RUN behave the same as that for consistency with the status quo?

Sounds good to me.